src/main/java/com/kodilla/ecommercee/controller/CartController.java

```diff
@@ -1,38 +1,71 @@
 package com.kodilla.ecommercee.controller;
 
-
+import com.kodilla.ecommercee.domain.Cart;
+import com.kodilla.ecommercee.domain.Order;
+import com.kodilla.ecommercee.domain.Product;
 import com.kodilla.ecommercee.domain.dto.CartDto;
+import com.kodilla.ecommercee.domain.dto.OrderDto;
 import com.kodilla.ecommercee.domain.dto.ProductDto;
+import com.kodilla.ecommercee.mapper.CartMapper;
+import com.kodilla.ecommercee.mapper.OrderMapper;
+import com.kodilla.ecommercee.mapper.ProductMapper;
+import com.kodilla.ecommercee.service.CartService;
+import com.kodilla.ecommercee.service.OrderService;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.web.bind.annotation.*;
 
-import java.util.ArrayList;
+import java.math.BigDecimal;
 import java.util.List;
+import java.util.stream.Collectors;
 
 @RestController
 @RequestMapping("/v1/ecommercee")
 public class CartController {
+    @Autowired
+    CartMapper cartMapper;
+    @Autowired
+    CartService cartService;
+    @Autowired
+    ProductMapper productMapper;
+    @Autowired
+    OrderMapper orderMapper;
+    @Autowired
+    OrderService orderService;
+    @Autowired
+    OrderController orderController;
 
     @RequestMapping(method = RequestMethod.POST, value = "newCart", consumes = "application/json")
     public void newCart(@RequestBody CartDto cartDto) {
-
+        cartService.saveCart(cartMapper.mapToCart(cartDto));
     }
 
     @RequestMapping(method = RequestMethod.GET, value = "getProducts")
-    public List<ProductDto> getProducts() {
-        return new ArrayList<>();
+    public List<ProductDto> getProducts(@RequestParam Long cartId) {
+        return productMapper.mapToProductDtoList(cartService.getCart(cartId).orElse(new Cart()).getProducts());
     }
 
     @RequestMapping(method = RequestMethod.PUT, value = "addToCart", consumes = "application/json")
-    public void addToCart(@RequestBody ProductDto productDto) {
-
+    public void addToCart(@RequestParam Long cartId, @RequestBody ProductDto productDto) {
+        Cart updatedCart = cartService.getCart(cartId).get();
+        updatedCart.getProducts().add(productMapper.mapToProduct(productDto));
+        cartService.saveCart(updatedCart);
     }
 
     @RequestMapping(method = RequestMethod.DELETE, value = "deleteFromCart")
-    public void deleteProduct(@RequestParam Long productId) {
-
+    public void deleteProduct(@RequestParam Long cartId, @RequestParam Long productId) {
```

Comment:
Czy delete powinien rzucać jakimś wyjątkiem jeśli nie znajdzie produktu (podanego id) w koszyku? W sumie to samo pytanie do wszystkich metod kontrolera gdzie podajemy jako parametr jakieś Id.

Reply:
throws NotFoundException {
...
.orElseThrow(NotFoundException::new));

Reply:
Wydaje mi się, że można dodać do usuwania we wszystkich controllerach :)

Reply:
Wrzuciłem do CartControllera, przy usuwaniu i wyszukiwaniu.